### PR TITLE
feat(server): add `fetchReportBySlug()`

### PR DESCRIPTION
**Pull request by baumstern**
Implement `fetchReportBySlug` function and streamline environment variable usage for fetching reports

### CODE REVIEW

File: app/impact-reports.server.ts [hunk 1]
```diff
@@ -17,13 +17,20 @@ let hypercertClient: HypercertClient | null = null;
...
+	}
+	const ownerAddress = process.env.HC_OWNER_ADDRESS;
+
 	try {
 		if (reports) {
 			console.log("Reports already exist, no need to fetch from remote");
 			console.log(`Existing reports: ${reports.length}`);
 		} else {
-			console.log("Fetching reports from remote");
+			console.log(
```
Comment (thebeyondr) [inline app/impact-reports.server.ts:31]: don't think we should be leaking the owner address via logs

**Review by thebeyondr — CHANGES_REQUESTED** [review]
don't think we should be leaking the owner address at all. We should limit console logs in general tbh. The core of the PR is good though. ready to approve once the minor tweaks are fixed

File: app/impact-reports.server.ts (hunk 1 above)
Comment (baumstern) [reply app/impact-reports.server.ts:31]: hmm, it's only showing up on the server side, so I don't think it'll be a problem, right?

Comment (thebeyondr) [reply app/impact-reports.server.ts:31]: Its registering to me as an attack vector for the app

>  so I don't think it'll be a problem

lol sounds like someone who's about to be hacked but doesn't know it yet

Comment (baumstern) [reply app/impact-reports.server.ts:31]: > Its registering to me as an attack vector for the app
> 
> > so I don't think it'll be a problem
> 
> lol sounds like someone who's about to be hacked but doesn't know it yet

I might be missing something here, but accessing the server essentially grants visibility into the source code and any embedded values. And address is not a sensitive information. But if you strongly feel it should be removed, I'm willing to do so

Comment (thebeyondr) [reply app/impact-reports.server.ts:31]: well if you dont think so, leggo

**Review by thebeyondr — APPROVED** [review]
no threats here lfg

**Pull request merged**